### PR TITLE
Prepare for Disabling MbsAPI

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
We want to remove MbsAPI.
So let's deprecate it first.

* A `BUILD_MBS` option already exists.
* Make sure, that setting it to OFF actually works.
* Do not yet set it to OFF
* Add `libdl` to `MCStepLogger`
* Move myself from CONTRIBUTORS to AUTHORS

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

**Comment by ChristianTackeGSI** [comment]
I have left out the "set BUILD_MBS default to OFF" for now.
I can add that commit soon here, or create a new PR after this one is merged.
What do you prefer?

File: CMakeLists.txt [hunk 1]
```diff
@@ -308,7 +308,9 @@ else()
   message(STATUS "eventdisplay will not be built, because ROOT has no opengl support.")
 endif()
 
-add_subdirectory(MbsAPI)
+if(BUILD_MBS)
```
Comment (dennisklein) [inline CMakeLists.txt:311]: Let's also declare this option as official build [option](https://cmake.org/cmake/help/latest/command/option.html) somewhere before this line.

**Review by dennisklein — CHANGES_REQUESTED** [review]
See comment, rest looks good!

**Comment by dennisklein** [comment]
> I have left out the "set BUILD_MBS default to OFF" for now.
> I can add that commit soon here, or create a new PR after this one is merged.
> What do you prefer?
> 
> Oh, and please add anyone in, who might be interested.

ah, yes, please set the default to `OFF` already.

File: CMakeLists.txt (hunk 1 above)
Comment (dennisklein) [reply CMakeLists.txt:311]: hm, seeing this, let's also rename it to `BUILD_MBSAPI` to match the component name.

Comment (ChristianTackeGSI) [reply CMakeLists.txt:311]: It's already an official `option`:
https://github.com/FairRootGroup/FairRoot/blob/dev/CMakeLists.txt#L295

So should we rename it, despite it already being "established"?
* Con: The name is already there, in theory people could already know about it and use `-DBUILD_MBS=ON` so be sure, they get this.
* Pro: It was broken anyway. (at least the OFF setting)

I don't have an opinion on this.

Comment (dennisklein) [reply CMakeLists.txt:311]: > So should we rename it, despite it already being "established"?

No, in that case, let's leave it. Sry, didn't remember this.

**Comment by dennisklein** [comment]
> An option BUILD_MBS is already available.
> Setting it to OFF breaks the build currently.

What breaks?

**Comment by ChristianTackeGSI** [comment]
> What breaks?

Phrased badly.
Better: "Without this PR".

So without this PR doing a `-DBUILD_MBS=OFF` did not work. Basically, because the examples were building, but Mbs did not.

That's why I added a bunch of more `if(BUILD_MBS)` around the place. To stop building any depenendencies of MBS.

**Review by dennisklein — APPROVED** [review]

**Pull request merged**